tests/sentry/explore/endpoints/test_explore_saved_query_detail.py

```diff
@@ -0,0 +1,373 @@
+        assert response.data["interval"] == "10m"
+
+    def test_put_query_without_access(self):
+        with self.feature(self.feature_name):
+            url = reverse(
+                "sentry-api-0-explore-saved-query-detail",
+                args=[self.org.slug, self.query_id_without_access],
+            )
+
+            response = self.client.put(
+                url,
+                {
+                    "name": "New query",
+                    "projects": self.project_ids,
+                    "range": "24h",
+                    "mode": "samples",
+                },
+            )
+
+            assert response.status_code == 404
+
+    def test_put_query_with_team(self):
+        team = self.create_team(organization=self.org, members=[self.user])
+        project = self.create_project(organization=self.org, teams=[team])
+        query = ExploreSavedQuery.objects.create(
+            organization=self.org,
+            created_by_id=self.user.id,
+            name="Test query",
+            query={"fields": ["span.op"], "mode": "samples"},
+        )
+        query.set_projects([project.id])
+
+        with self.feature(self.feature_name):
+            url = reverse(
+                "sentry-api-0-explore-saved-query-detail",
+                args=[self.org.slug, query.id],
+            )
+
+            response = self.client.put(
+                url, {"name": "New query", "projects": [], "range": "24h", "mode": "samples"}
+            )
+
+            assert response.status_code == 200
+
+    def test_put_query_without_team(self):
+        team = self.create_team(organization=self.org, members=[])
+        project = self.create_project(organization=self.org, teams=[team])
+        query = ExploreSavedQuery.objects.create(
+            organization=self.org,
+            created_by_id=self.user.id,
+            name="Test query",
+            query={"fields": ["span.op"], "mode": "samples"},
+        )
+        query.set_projects([project.id])
+
+        with self.feature(self.feature_name):
+            url = reverse(
+                "sentry-api-0-explore-saved-query-detail",
+                args=[self.org.slug, query.id],
+            )
+
+            response = self.client.put(url, {"name": "New query", "projects": [], "range": "24h"})
+
+            assert response.status_code == 400
+            assert "No Projects found, join a Team" == response.data["detail"]
+
+    def test_put_org_without_access(self):
+        with self.feature(self.feature_name):
+            url = reverse(
+                "sentry-api-0-explore-saved-query-detail",
+                args=[self.org_without_access.slug, self.query_id],
+            )
+            response = self.client.put(
+                url, {"name": "New query", "projects": self.project_ids, "range": "24h"}
+            )
+
+        assert response.status_code == 403, response.content
+
+    def test_put_disallow_when_no_project_access(self):
```

Comment:
Not entirely sure if this case is handled, but I noticed the setup queries we always set up the projects. Is this required? Trace explorer automatically selects the first project for me but I can unselect it to get "all projects". Just wondering if I can save a query with that.